[windows] Electron application does not start after installing it

After #738 I get this when I start the electron application after the installation
```
Failed to start the backend application.
Error: %1 is not a valid Win32 application.
\\?\C:\Users\kittaakos\AppData\Local\Programs\theia-electron-app-example\resources\app\node_modules\node-pty\build\Release\pty.node
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:173:20)
    at Object.Module._extensions..node (module.js:598:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:173:20)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (C:\Users\kittaakos\AppData\Local\Programs\theia-electron-app-example\resources\app\node_modules\node-pty\lib\windowsPtyAgent.js:9:11)
    at Object.<anonymous> (C:\Users\kittaakos\AppData\Local\Programs\theia-electron-app-example\resources\app\node_modules\node-pty\lib\windowsPtyAgent.js:160:3)
C:\Users\kittaakos\AppData\Local\Programs\theia-electron-app-example\resources\app\node_modules\@theia\core\lib\node\cluster\main.js:41
    throw reason;
    ^

Error: %1 is not a valid Win32 application.
\\?\C:\Users\kittaakos\AppData\Local\Programs\theia-electron-app-example\resources\app\node_modules\node-pty\build\Release\pty.node
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:173:20)
    at Object.Module._extensions..node (module.js:598:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:173:20)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (C:\Users\kittaakos\AppData\Local\Programs\theia-electron-app-example\resources\app\node_modules\node-pty\lib\windowsPtyAgent.js:9:11)
    at Object.<anonymous> (C:\Users\kittaakos\AppData\Local\Programs\theia-electron-app-example\resources\app\node_modules\node-pty\lib\windowsPtyAgent.js:160:3)
The express server worker 1 has been disconnected.
The express server worker 1 has been stopped.
```

For some reason, editor.main module is not included:
```
loader.js:7 Loading "vs/editor/editor.main" failed
function.r.onError.r.onError @ loader.js:7
r.onError @ loader.js:7
s._onLoadError @ loader.js:7
s @ loader.js:7
(anonymous) @ loader.js:7
e.triggerErrorback @ loader.js:7
(anonymous) @ loader.js:7
(anonymous) @ loader.js:7
readFileAfterOpen @ fs.js:359
loader.js:7 Detail:  Error: ENOENT: no such file or directory, open 'c:\Users\kittaakos\dev\theia-electron-app-example\dist\win-unpacked\c%3A\Users\kittaakos\dev\theia-electron-app-example\dist\win-unpacked\resources\app\lib\vs\editor\editor.main.js'
function.r.onError.r.onError @ loader.js:7
r.onError @ loader.js:7
s._onLoadError @ loader.js:7
s @ loader.js:7
(anonymous) @ loader.js:7
e.triggerErrorback @ loader.js:7
(anonymous) @ loader.js:7
(anonymous) @ loader.js:7
readFileAfterOpen @ fs.js:359
loader.js:7 Error: ENOENT: no such file or directory, open 'c:\Users\kittaakos\dev\theia-electron-app-example\dist\win-unpacked\c%3A\Users\kittaakos\dev\theia-electron-app-example\dist\win-unpacked\resources\app\lib\vs\editor\editor.main.js'
function.r.onError.r.onError @ loader.js:7
r.onError @ loader.js:7
s._onLoadError @ loader.js:7
s @ loader.js:7
(anonymous) @ loader.js:7
e.triggerErrorback @ loader.js:7
(anonymous) @ loader.js:7
(anonymous) @ loader.js:7
readFileAfterOpen @ fs.js:359
loader.js:7 Here are the modules that depend on it:
function.r.onError.r.onError @ loader.js:7
r.onError @ loader.js:7
s._onLoadError @ loader.js:7
s @ loader.js:7
(anonymous) @ loader.js:7
e.triggerErrorback @ loader.js:7
(anonymous) @ loader.js:7
(anonymous) @ loader.js:7
readFileAfterOpen @ fs.js:359
loader.js:7 ["===anonymous1==="]0: "===anonymous1==="length: 1__proto__: Array(0)
function.r.onError.r.onError @ loader.js:7
r.onError @ loader.js:7
s._onLoadError @ loader.js:7
s @ loader.js:7
(anonymous) @ loader.js:7
e.triggerErrorback @ loader.js:7
(anonymous) @ loader.js:7
(anonymous) @ loader.js:7
readFileAfterOpen @ fs.js:359

```

```typescript
/C:/Users/Marc/theia/examples/electron/lib/vs/loader.js:7 Detail:  Error: ENOENT: no such file or directory, open 'C:\Users\Marc\theia\examples\electron\c%3A\Users\Marc\theia\examples\electron\lib\vs\editor\editor.main.js'
function.r.onError.r.onError @ /C:/Users/Marc/theia/examples/electron/lib/vs/loader.js:7
/C:/Users/Marc/theia/examples/electron/lib/vs/loader.js:7 Error: ENOENT: no such file or directory, open 'C:\Users\Marc\theia\examples\electron\c%3A\Users\Marc\theia\examples\electron\lib\vs\editor\editor.main.js'
function.r.onError.r.onError @ /C:/Users/Marc/theia/examples/electron/lib/vs/loader.js:7
/C:/Users/Marc/theia/examples/electron/lib/vs/loader.js:7 Here are the modules that depend on it:
function.r.onError.r.onError @ /C:/Users/Marc/theia/examples/electron/lib/vs/loader.js:7
/C:/Users/Marc/theia/examples/electron/lib/vs/loader.js:7 Array(1)
function.r.onError.r.onError @ /C:/Users/Marc/theia/examples/electron/lib/vs/loader.js:7
```

--- a/packages/monaco/src/electron-browser/monaco-electron-module.ts
+++ b/packages/monaco/src/electron-browser/monaco-electron-module.ts
@@ -5,17 +5,30 @@
  * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
  */
 
+import * as path from "path";
 import { ContainerModule } from "inversify";
-import { FileUri } from "@theia/core/lib/node/file-uri";
 import { loadVsRequire, loadMonaco } from "../browser/monaco-loader";
 
 export { ContainerModule };
 
 const s = <any>self;
 
+/**
+ * We cannot use `FileUri#create` because URIs with file scheme cannot be properly decoded via the AMD loader.
+ * So if you have a FS path on Windows: `C:\Users\foo`, then you will get a URI `file:///c%3A/Users/foo` which
+ * will be converted into the `c%3A/Users/foo` FS path on Windows by the AMD loader.
+ */
+const uriFromPath = (filePath: string) => {
+    let pathName = path.resolve(filePath).replace(/\\/g, '/');
+    if (pathName.length > 0 && pathName.charAt(0) !== '/') {
+        pathName = '/' + pathName;
+    }
+    return encodeURI('file://' + pathName);
+};
+
 export default loadVsRequire(global)
     .then(vsRequire => {
-        const baseUrl = FileUri.create(__dirname).toString();
+        const baseUrl = uriFromPath(__dirname);
         vsRequire.config({ baseUrl });
 
         // workaround monaco-css not understanding the environment

